contracts/RewardPool.sol

```diff
@@ -80,15 +76,18 @@ contract RewardPool is Initializable, Versionable, Ownable, ReentrancyGuard {
     emit Setup(_tally, _rewardManager, _tokenManager);
   }
 
-  function submitPayeeMerkleRoot(bytes32 payeeRoot)
-    external
-    onlyTally
-    returns (bool)
-  {
-    payeeRoots[numPaymentCycles] = payeeRoot;
+  function submitPayeeMerkleRoot(
+    address rewardProgramID,
+    uint256 paymentCycle,
```

Comment:
I'd say remove `paymentCycle` here. And keep track of it inside the contract.

Reply:
So I was just pondering this. This may be a guard from double-writing merkle roots. The paymentCycle might act like a nonce.